### ISSUE TITLE
Userstream for binance futures doesn't connect

### ISSUE THREAD

**Issue by hbizot** [oliver-zehentleitner/unicorn-binance-websocket-api#35]
Check this or we will delete your issue. (fill in the checkbox with an X like so: [x])
- [x] I have searched for other issues with the same problem or similar feature requests. 

#### Select one:
- [x ] Bug
- [ ] Feature Request
- [ ] Technical Help
- [ ] Other

#### Environment
- [ ] Are you using the module on a VPS or other Cloud hosting?
- [ ] If so, have you tried running the code on your local machine?
- [ ] Are you using the module on a Raspberry Pi?

#### Operating System? (include version)
- [x] macOS
- [ ] Windows
- [ ] Linux (include flavour)

#### Python Version Requirement
- [x] I am using Python 3.5.3 or above

#### Exact Python Version?
3.6.3```
Include here the response of 'python --version' or 'python3 --version'
```

#### Pip Version?
```
Include here the response of 'pip --version' or 'pip3 --version'
```
pip 19.3.1 

#### Dependencies
Run `pip list > pip_list.txt` and upload the file.

#### UNICORN Binance WebSocket API Version?
```
Did you upgrade to the latest release version with `pip install unicorn-binance-websocket-api --upgrade`?

Please control what version you are using with this script and post the output: https://github.com/oliver-zehentleitner/unicorn-binance-websocket-api/blob/master/tools/get_used_module_version.py
```

#### Description of your issue

The user stream for binance futures, doesn't seem to be working

I think in the methods: delete_listen_key, get_listen_key, keepalive_listen_key there is something missing.

You are referring to a path: path = "/api/v1/userDataStream"

But it seems that this path should be different for spot/margin or futures

For futures it should be: /fapi/v1/listenKey
For margin it should be: /sapi/v1/userDataStream

at least according to:
https://binance-docs.github.io/apidocs/futures/en/#user-data-streams
https://binance-docs.github.io/apidocs/spot/en/#user-data-streams

I created an instance like this:
# create instances of BinanceWebSocketApiManager
binance_futures_websocket_api_manager = BinanceWebSocketApiManager(exchange="binance.com-futures")

The log-file says this:
2019-11-11 14:00:05,980 [CRITICAL] 8052 123145426292736 unicorn_binance_websocket_api_connection: BinanceWebSocketApiConnection->await._conn.__aenter__(ce167999-eaf1-4b28-900a-ef4aaafa0ef9, arr, !userData) - Probably no internet connection?
2019-11-11 14:00:05,980 [CRITICAL] 8052 123145426292736 unicorn_binance_websocket_api_manager: BinanceWebSocketApiManager->stream_is_crashing(ce167999-eaf1-4b28-900a-ef4aaafa0ef9)


Thanks.

**Comment by oliver-zehentleitner** [comment]
Thanks for the detailed report!
I will fix it soon!
Best regards,
Oliver

**Comment by oliver-zehentleitner** [comment]
hi hbizot!

i made a few changes to support the margin and futures endpoint and i updated the path to the spot endpoint: https://github.com/oliver-zehentleitner/unicorn-binance-websocket-api/commit/e256f84afcd110ed2f0235bf8570640eea6ac098

i can not test it! would you be so kind and test the update?

to install it you can use pip:
`pip install https://github.com/oliver-zehentleitner/unicorn-binance-websocket-api/tarball/master --upgrade`

i think what now is "spot" is the default userData stream, before `/api/v1/userDataStream` and now `/api/v3/userDataStream` and can be used with `exchange="binance.com"`

to use the different userData streams start the manager with this options:
Binance: BinanceWebSocketApiManager(exchange="binance.com")
Binance: Margin BinanceWebSocketApiManager(exchange="binance.com-margin")
Binance: Futures BinanceWebSocketApiManager(exchange="binance.com-futures")

best regards,
Oliver

**Comment by hbizot** [comment]
It works like a charm.
I only tested binance-futures btw.

Thanks

**Comment by oliver-zehentleitner** [comment]
Cool! Your welcome!

I tested the spot userData streams, works also very well!

**Comment by oliver-zehentleitner** [comment]
I released this code as 1.9.0 and pushed it to pypi: https://pypi.org/project/unicorn-binance-websocket-api/
